### PR TITLE
Dev Diary June 2017

### PR DESCRIPTION
**Pull request by cmontella**

### CODE REVIEW

File: _posts/2017-07-06-June.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+layout: post
+title: "Dev Diary (June 2017)"
+author: "Corey Montella"
+tags: []
+---
+
+## Preview 5 of v0.3
+
+We released a new preview of v0.3, that includes several bug fixes and some new features. For a full changelog, see the [release notes](http://incidentalcomplexity.com/2017/06/28/release-0.3p5/). Two of the new features, the `console` and `file` libraries, open Eve up to using Eve in new ways.
```
Comment (joshuafcole) [inline _posts/2017-07-06-June.md]: "open Eve up to using Eve in new ways"

File: _posts/2017-07-06-June.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+title: "Dev Diary (June 2017)"
+author: "Corey Montella"
+tags: []
+---
+
+## Preview 5 of v0.3
+
+We released a new preview of v0.3, that includes several bug fixes and some new features. For a full changelog, see the [release notes](http://incidentalcomplexity.com/2017/06/28/release-0.3p5/). Two of the new features, the `console` and `file` libraries, open Eve up to using Eve in new ways. 
+
+We've supported running Eve in headless mode (without hooks to the browser) for a while now, but it wasn't useful since most of our UI, debugging, and IO at the moment are done through the browser. The `console` library allows you to write to `sdtout` and `stderr`, so you can now pipe values generated by Eve to other applications (we still need to add the ability to pipe values _into_ Eve, but that's on our list).
```
Comment (joshuafcole) [inline _posts/2017-07-06-June.md]: "sdtout" -> "stdout"

File: _posts/2017-07-06-June.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+tags: []
+---
+
+## Preview 5 of v0.3
+
+We released a new preview of v0.3, that includes several bug fixes and some new features. For a full changelog, see the [release notes](http://incidentalcomplexity.com/2017/06/28/release-0.3p5/). Two of the new features, the `console` and `file` libraries, open Eve up to using Eve in new ways. 
+
+We've supported running Eve in headless mode (without hooks to the browser) for a while now, but it wasn't useful since most of our UI, debugging, and IO at the moment are done through the browser. The `console` library allows you to write to `sdtout` and `stderr`, so you can now pipe values generated by Eve to other applications (we still need to add the ability to pipe values _into_ Eve, but that's on our list). 
+
+Speaking of getting data into Eve, the `file` library allows you write and read files directly from the filesystem, opening up the ability to read larger datasets into Eve. We'll be shortly adding facilities in Eve that parse common data formats into records, like JSON and CSS for starters.
```
Comment (joshuafcole) [inline _posts/2017-07-06-June.md]: "JSON and CSS" -> "JSON and CSV"

Comment (joshuafcole) [reply _posts/2017-07-06-June.md]: Drop "for starters"

File: _posts/2017-07-06-June.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Community
+
+### Community Slack
+
+We opened a [community slack channel](https://slack-signup.witheve.com/), so you can chat with and get help from us and fellow Eve developers. There's usually at least one dev hanging around the channel during working hours (Pacific time), but in the future we're planning explicit office hours where our sole purpose is to help you solve problems with your code. For now, we're keeping it casual so drop by and we can talk about Eve.
+
+### Tweve
+
+Brian Theado has been working on [Tweve](https://btheado.github.io/tweve/), which marries Eve and Tiddlywiki. Tiddlywiki is a notebook-like environment that allows you to embed rich-content in your notes. Tweve is a plugin for Tiddlywiki that allows you to embed Eve code in your notebook. One of the more interesting parts of this project is that it allows you to embed multiple Eve programs in a single document, a feature that we demoed internally but never released. We're very interested to see how far Brian takes his project!
```
Comment (joshuafcole) [inline _posts/2017-07-06-June.md]: "multiple programs in a single document" AND ALSO "multiple documents as a single program"

Comment (joshuafcole) [reply _posts/2017-07-06-June.md]: The last sentence feels off-putting. Maybe instead encourage users interested in tiddlywiki to give it a shot or contribute to it's development?

File: _posts/2017-07-06-June.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+---
+layout: post
+title: "Dev Diary (June 2017)"
+author: "Corey Montella"
+tags: []
+---
+
+## Preview 5 of v0.3
+
+We released a new preview of v0.3, that includes several bug fixes and some new features. For a full changelog, see the [release notes](http://incidentalcomplexity.com/2017/06/28/release-0.3p5/). Two of the new features, the `console` and `file` libraries, provide new options for using Eve.
```
Comment (ibdthor) [inline _posts/2017-07-06-June.md]: no comma after "v0.3"

Comment (ibdthor) [reply _posts/2017-07-06-June.md]: "please see the release notes"

File: _posts/2017-07-06-June.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+We've supported running Eve in headless mode (without hooks to the browser) for a while now, but it wasn't useful since most of our UI, debugging, and IO at the moment are done through the browser. The `console` library allows you to write to `stdout` and `stderr`, so you can now pipe values generated by Eve to other applications (we still need to add the ability to pipe values _into_ Eve, but that's on our list).
```
Comment (ibdthor) [inline _posts/2017-07-06-June.md]: "awhile", one word

File: _posts/2017-07-06-June.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+tags: []
+---
+
+## Preview 5 of v0.3
+
+We released a new preview of v0.3, that includes several bug fixes and some new features. For a full changelog, see the [release notes](http://incidentalcomplexity.com/2017/06/28/release-0.3p5/). Two of the new features, the `console` and `file` libraries, provide new options for using Eve.
+
+We've supported running Eve in headless mode (without hooks to the browser) for a while now, but it wasn't useful since most of our UI, debugging, and IO at the moment are done through the browser. The `console` library allows you to write to `stdout` and `stderr`, so you can now pipe values generated by Eve to other applications (we still need to add the ability to pipe values _into_ Eve, but that's on our list). 
+
+Speaking of getting data into Eve, the `file` library allows you write and read files directly from the filesystem, opening up the ability to read larger datasets into Eve. We'll be shortly adding facilities in Eve that parse common data formats into records, like JSON and CSV.
```
Comment (ibdthor) [inline _posts/2017-07-06-June.md]: "adding facilities shortly", more typical adverb syntax

File: _posts/2017-07-06-June.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+## Progress with v0.4
+
+Many of you have asked about the status of v0.4, and we're pleased to say it's coming along nicely. For those who don't know, the v0.4 runtime is implemented in Rust, mainly for increased performance. It's similar conceptually to the v0.3 runtime (based on the research of Frank McSherry into differential dataflow), but is designed with the lessons learned from v0.3. We'll share more about performance when the runtime is feature-complete and ready for wider distribution.
```
Comment (ibdthor) [inline _posts/2017-07-06-June.md]: "research of Frank McSherry in"

File: _posts/2017-07-06-June.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+## Community
+
+### Community Slack
+
+We opened a [community slack channel](https://slack-signup.witheve.com/), so you can chat with and get help from us and fellow Eve developers. There's usually at least one dev hanging around the channel during working hours (Pacific time), but in the future we're planning explicit office hours where our sole purpose is to help you solve problems with your code. For now, we're keeping it casual so drop by and we can talk about Eve.
```
Comment (ibdthor) [inline _posts/2017-07-06-June.md]: "For now we're keeping it casual, so drop by..."

File: _posts/2017-07-06-June.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+### Tweve
+
+Brian Theado has been working on [Tweve](https://btheado.github.io/tweve/), which marries Eve and Tiddlywiki. Tiddlywiki is a notebook-like environment that allows you to embed rich-content in your notes. Tweve is a plugin for Tiddlywiki that allows you to embed Eve code in your notebook. One of the more interesting parts of this project is that it allows you to embed multiple Eve programs in a single document, a feature that we demoed internally but never released. We're very interested to see how far Brian takes his project!
```
Comment (ibdthor) [inline _posts/2017-07-06-June.md]: is "rich-content" supposed to be hyphenated?

File: _posts/2017-07-06-June.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+### Eve Around the World
+
+**Copenhagen, June 27**
+
+Zubair Quraishi held yet another successful meetup in Copenhagen, and we're definitely grateful for his continued evangelism across the pond! He's posted some pictures and reactions on the [Meetup website](https://www.meetup.com/evecph/events/240055666/) and the [mailing list](https://groups.google.com/d/msg/eve-talk/PwlgSIdppMQ/Xb9zI5niAAAJ). Thanks again Zubair!
+
+**San Francisco June 27**
```
Comment (ibdthor) [inline _posts/2017-07-06-June.md]: "San Francisco, June 27"

**Review by ibdthor — CHANGES_REQUESTED** [review]

File: _posts/2017-07-06-June.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+## Community
+
+### Community Slack
+
+We opened a [community slack channel](https://slack-signup.witheve.com/), so you can chat with and get help from us and fellow Eve developers. There's usually at least one dev hanging around the channel during working hours (Pacific time), but in the future we're planning explicit office hours where our sole purpose is to help you solve problems with your code. For now we're keeping it casual, so drop by and we can talk about Eve.
+
+### Tweve
+
+Brian Theado has been working on [Tweve](https://btheado.github.io/tweve/), which marries Eve and Tiddlywiki. Tiddlywiki is a notebook-like environment that allows you to embed rich-content in your notes. Tweve is a plugin for Tiddlywiki that allows you to embed Eve code in your notebook. One of the more interesting parts of this project is that it allows you to embed multiple Eve programs in a single document, as well as multiple documents as a single program (a feature that we demoed internally but never released). We're very interested to see how far Brian takes his project!
```
Comment (ibdknox) [inline _posts/2017-07-06-June.md]: remove:
> (a feature that we demoed internally but never released)

File: _posts/2017-07-06-June.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+### Eve Around the World
+
+**Copenhagen, June 27**
+
+Zubair Quraishi held yet another successful meetup in Copenhagen, and we're definitely grateful for his continued evangelism across the pond! He's posted some pictures and reactions on the [Meetup website](https://www.meetup.com/evecph/events/240055666/) and the [mailing list](https://groups.google.com/d/msg/eve-talk/PwlgSIdppMQ/Xb9zI5niAAAJ). Thanks again Zubair!
+
+**San Francisco, June 27**
+
+Thanks to everyone who came to our second meetup, in which we helped people install Eve and introduced the language to people who were new. We learned a lot by watching people go through the quickstart tutorial, which has caused us to re-evaluate the getting started process. We don't have a third meetup planned at this time, but we'll update the blog when we do!
```
Comment (ibdknox) [inline _posts/2017-07-06-June.md]: there's no reason to say we don't have a plan, let's just... have a plan.

**Review by ibdthor — APPROVED** [review]

**Review by ibdthor — APPROVED** [review]

**Review by ibdthor — APPROVED** [review]

**Review by ibdknox — APPROVED** [review]

**Review by joshuafcole — APPROVED** [review]
:+1: :pizza: 📦

**Pull request merged**